src/garage/_dtypes.py

```diff
@@ -669,6 +707,10 @@ class TimeStepBatch(
         step_types (numpy.ndarray): A numpy array of `StepType with shape (
             batch_size,) containing the time step types for all transitions in
             this batch.
+        episode_infos (dict): A dict of episode-level information.
```

Comment:
please include the shape information and validate below. for a TimeStepBatch, we need episode_info for each TimeStep, because the TimeSteps don't need to share a "parent" episode (i.e. unlike EpisodeBatch, we will need a copy for each TimeStep).

using the terminology from above, this is :math:`(N, S^*)`